recipes-openxt/openxt/openxt-measuredlaunch/ml-functions

```diff
@@ -128,46 +111,29 @@ hash_extend() {
     local second=$2
     local buffer="/tmp/.buffer"
 
-    is_tpm_2_0
-    local tpm2=$?
-
-    if [ "${tpm2}" -eq 0 ]; then
-        default_first="0000000000000000000000000000000000000000000000000000000000000000"
-        shaalg="sha256sum"
-        size=64
-    else
-        default_first="0000000000000000000000000000000000000000"
-        shaalg="sha1sum"
-        size=40
-    fi
+    [[ $first == 0 ]] && first="0000000000000000000000000000000000000000"
+    [[ $(echo -n $first|wc -c) -ne 40 ]] && return 1
+    [[ $(echo -n $second|wc -c) -ne 40 ]] && return 1
 
-    [[ $first == 0 ]] && first=${default_first}
-    [[ $(echo -n $first|wc -c) -ne ${size} ]] && return 1
-    [[ $(echo -n $second|wc -c) -ne ${size} ]] && return 1
     echo -n $first | hex2bin >$buffer
     echo -n $second | hex2bin >>$buffer
 
-    echo $(${shaalg} $buffer | awk '{ print $1 }')
+    echo $(sha1sum $buffer | awk '{ print $1 }')
     rm $buffer
 }
 
 # Function to calculate PCR 17
 # return 0 on success
 # return 1 on failure
 pcr17() {
-    is_tpm_2_0
-    local tpm2=$?
+    local root="${1:-/}"
+    local pcr17=''
 
-    if [ "${tpm2}" -eq 0 ]; then
-        pcr17=$(tpm2_listpcrs -g 0xb | grep PCR_17 | cut -f2 -d: | tr -d ' ' | tr -d '\n')
```

Comment:
I was going to strip this out because the new approach does not need these functions. I added it back in for forward seal when coming up from a stable-6 box. If you are doing measured launch in a stable-6 environment, then have to be on a tpm1.2. If you are upgrading from a system that has since incorporated then tboot/xen/linux trio patch to export the log, this function is not used.